mount host /boot into cilium-agent container

agrevtcev opened the pull request:
mount host /boot into cilium-agent container, as it required to run `bpftool -j feature probe` during kernel feature probing

Fixes: #20981

Signed-off-by: Alexey Grevtcev [alexey@grevtcev.ru](mailto:alexey@grevtcev.ru)

gandro commented on install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml:
```diff
@@ -308,6 +308,11 @@ spec:
         - mountPath: /host/proc/sys/kernel
           name: host-proc-sys-kernel
         {{- end}}
+        # Required to run 'bpftool -j feature probe'
```
Also, since this is disabled by default, can we make both added sections conditional on `{{ if .Values.bpfClockProbe }}`? Thanks!

aanm replied:
@gandro what is disabled by default? I would suggest to enable mounting `/boot` by default with a helm option to disabling it if necessary.

gandro replied:
My understanding is that access to /boot is only needed if (and only if) `.Values.bpfClockProbe=true`.  That's why I would make it conditional on that.

If `.Values.bpfClockProbe` is false (which it seems to be by default on recent versions), then there is no point in mounting /boot.

gandro replied:
Oh, I just checked the source and it seems the check that requires /boot is done unconditionally (i.e. even if bpfClockProbe is false). In which case my point is moot, please ignore:

https://github.com/cilium/cilium/blob/27700b02032af349b67d97789575a1533466686a/daemon/cmd/daemon_main.go#L2050

gandro replied:
@aanm You requested changes. What changes are you requesting? Adding a independent Helm flag to make the boot mount conditional?

aanm replied:
> @aanm You requested changes. What changes are you requesting? Adding a independent Helm flag to make the boot mount conditional?

@gandro correct.

agrevtcev replied:
@aanm are you sure it's necessary? It seems we are running initClockSourceOption() unconditionally, despite the .Values.bpfClockProbe value.
https://github.com/cilium/cilium/blob/27700b02032af349b67d97789575a1533466686a/daemon/cmd/daemon_main.go#L1413

aanm replied:
@agrevtcev yes, not everyone might want to mount `/boot` into Cilium DaemonSet even if it is read-only so it does make sense to mark it as optional. That function is running uncoditionally but we can do as a follow up to add a flag to specify that value for users that don't want to mount `/boot`

aanm reviewed (changes requested):
@agrevtcev please see https://github.com/cilium/cilium/pull/21113/files#r958447909

aanm reviewed (changes requested):
Changes LGTM, just please squash all commits into one. Thank you!

kaworu reviewed (changes requested):
Thanks for the update @agrevtcev! The diff LGTM, but CI is complaining for a couple of things:

1. Please run `Documentation/update-spelling_wordlist.sh hostBoot mountHostBoot` to fix [this failure](https://github.com/cilium/cilium/runs/8112924823?check_suite_focus=true),
2. Replace the commit message's `Signed-off-by` line with `Signed-off-by: Alexey Grevtcev <alexey@grevtcev.ru>` (notice the `<` and `>` enclosing the email address) to fix [this failure](https://github.com/cilium/cilium/runs/8112921045?check_suite_focus=true),
3. Finally, rebase and squash your changes into one commit.

Thank you! :pray:

agrevtsev committed:
Mount host /boot into cilium-agent container

It's required to run 'bpftool -j feature probe'
during kernel feature probing.
Control host /boot mount with values,
allow to override host /boot path.

Fixes: #20981

Signed-off-by: Alexey Grevtcev <alexey@grevtcev.ru>

aanm reviewed (approved):
Thank you!

aanm reviewed (approved):
Thank you very much!
